src/evaluate.cpp

```diff
@@ -887,14 +889,9 @@ namespace {
 #ifdef THREECHECK
             if (pos.is_three_check())
             {
-                switch(pos.checks_given(Them))
-                {
-                case CHECKS_NB:
-                case CHECKS_3:
-                case CHECKS_2:  kingDanger += 2 * kingDanger; break;
-                case CHECKS_1:  kingDanger += kingDanger; break;
-                case CHECKS_0:  kingDanger += kingDanger / 2; break;
-                }
+              int i = std::max(0, 2 - pos.checks_given(Them));
+              if (i < 3)
```

Comment:
In case there would be support for, e.g. 5-check chess. Naturally, then, SF would count checks remaining until loss, not already given: it then could be any positive integer (This was suggested by H.G.Muller). Then 3-check or 5-check are determined by the starting position.